src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -487,12 +471,7 @@ private TypeSymbolWithAnnotations SubstituteType(AbstractTypeMap typeMap, Func<A
 
                 if (newIsNullable == true)
                 {
-                    if (newCustomModifiers.IsEmpty)
-                    {
-                        return new NullableReferenceTypeWithoutCustomModifiers(newTypeWithModifiers.TypeSymbol);
-                    }
-
-                    return new NullableReferenceTypeWithCustomModifiers(newTypeWithModifiers.TypeSymbol, newCustomModifiers);
+                    return new NonLazyType(newTypeWithModifiers.TypeSymbol, isNullable: true, newCustomModifiers);
```

Comment:
Nit: It's a bit odd that other cases above use `Create`, but this case directly calls the constructor.